src/Npgsql/Internal/NpgsqlReadBuffer.cs

```diff
@@ -576,14 +577,17 @@ public double ReadDouble()
         return result;
     }
 
-    [Conditional("DEBUG")]
-    unsafe void CheckBounds<T>() where T : unmanaged
+    void CheckBounds(int count)
```

Comment:
As we're including this in release builds now I refactored it to be non-generic, for size reasons.